src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -1094,6 +1129,13 @@ void Replayer<I>::handle_copy_image_progress(uint64_t object_number,
     object_number, object_count);
   m_local_object_count = object_count;
 
+  if (g_perf_counters) {
```

Comment:
For me it would make more sense if for snapshot based mirroring we counted snapshots (not objects). The counter would be more useful to me. 

But, yes, aggregating the snapshot based counters with the journal counters does not make much sense. I would introduce new `rbd_mirror_snapshot_replay_*` counters.

Ideally, I would also rename the old `rbd_mirror_replay_*` counters to `rbd_mirror_journal_replay_*`, but I am not sure if users would be happy with this change.

Reply:
Ok yes I thought that if you had no diff between two snapshot the copied objects would be 0 but that's not the case so I will follow your thought and only do a snapshot count

Reply:
I also separated the global perf counter of the journal and the snapshot replayer.